waverecoverygui.py

```diff
@@ -0,0 +1,340 @@
+        form_layout.addRow(self.sample_rate_label, self.sample_rate_spinbox)
+        form_layout.addRow(self.bits_per_sample_label, self.bits_per_sample_combobox)
+        form_layout.addRow(self.channels_label, self.channels_combobox)
+        form_layout.addWidget(self.force_checkbox)
+        form_layout.addRow(self.application_label, self.application_combo)
+        form_layout.addRow(self.use_offset_checkbox)
+        form_layout.addRow(self.offset_label, self.offset_spinbox)
+        form_layout.addRow(self.end_offset_label, self.end_offset_spinbox)
+        form_layout.addWidget(self.version_checkbox)
+        form_layout.addWidget(self.verbose_checkbox)
+
+        # Apply styles to buttons and "Browse" buttons
+        button_style = '''
+            QPushButton, QTextEdit {
+                background-color: #007ACC;
+                color: white;
+                border: none;
+                border-radius: 5px;
+                padding: 8px 16px;
+            }
+
+            QPushButton:hover, QTextEdit:hover {
+                background-color: #005F99;
+            }
+        '''
+
+        self.restore_button = QPushButton('Restore')
+        self.restore_button.clicked.connect(self.restore)
+        self.restore_button.setStyleSheet(button_style)  # Apply the style to the button
+        form_layout.addRow(self.restore_button)
+
+        self.browse_source_button.setStyleSheet(button_style)  # Apply the style to the "Browse" button
+        self.browse_dest_button.setStyleSheet(button_style)  # Apply the style to the "Browse" button
+
+        self.cancel_all_button = QPushButton('Cancel All')
+        self.cancel_all_button.clicked.connect(self.cancel_all_threads)
+        self.cancel_all_button.setStyleSheet(button_style)  # Apply the style to the button
+        form_layout.addRow(self.cancel_all_button)
+
+        self.progress_bar = QProgressBar()
+        self.progress_bar.setStyleSheet("QProgressBar::chunk { background-color: green; } QProgressBar { text-align: center; color: white; border-radius: 10px; }")
+        form_layout.addRow(self.progress_bar)
+
+        self.log_box = QTextEdit()
+        self.log_box.setReadOnly(True)
+        form_layout.addRow(self.log_box)
+
+        self.central_widget.setLayout(form_layout)
+
+    def toggle_offset_fields(self):
+        enabled = self.use_offset_checkbox.isChecked()
+        self.offset_label.setEnabled(enabled)
+        self.offset_spinbox.setEnabled(enabled)
+        self.end_offset_label.setEnabled(enabled)
+        self.end_offset_spinbox.setEnabled(enabled)
+
+    def browse_source(self):
+        source_path = QFileDialog.getExistingDirectory(self, 'Select Source Directory')
+        self.source_path_text.setPlainText(source_path)
+
+    def browse_dest(self):
+        dest_path = QFileDialog.getExistingDirectory(self, 'Select Destination Directory')
+        self.dest_path_text.setPlainText(dest_path)
+
+    def restore(self):

```

Comment:
When clicking restore nothing happens for me, except the progress being set to 100%. But the wave recovery was actually not invoked for me. Do you also experience this behavior? Has to be debugged I guess

Reply:
i think your file poblem, my sample here look fine: 

https://gofile.io/d/Afa0rx

![2024-04-17_201251](https://github.com/david-pace/wave-recovery-tool/assets/85211068/6505932e-e8a0-4ecd-98ca-9a315bff4959)

Reply:
I finally found a solution that also works on my system. I had to use a different thread management approach. I am going to merge your pull request and will apply the fixes afterwards.

Thanks again for your contribution and also for your patience.